optax/contrib/_ademamix.py

```diff
@@ -0,0 +1,182 @@
+"""AdeMAMix.
+
+Implementation of
+"THE ADEMAMIX OPTIMIZER: BETTER, FASTER, OLDER"
+(https://arxiv.org/pdf/2409.03137) by Matteo Pagliardini, 
+Pierre Ablin and David Grangier.
+"""
+import optax.tree_utils as otu
+from typing import NamedTuple, Optional
+import chex
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from optax._src import base
+from optax._src import combine
+from optax._src import numerics
+from optax._src import transform
+
+
+class ScaleByAdemamixState(NamedTuple):
+  """State for the Ademamix algorithm."""
+
+  count: chex.Array
+  count_m2: chex.Array
+  m1: base.Updates
+  m2: base.Updates
+  nu: base.Updates
+
+
+def scale_by_ademamix(
+  b1: float = 0.9,
+  b2: float = 0.999,
+  b3: float = 0.9999,
+  alpha: float = 5.0,
+  b3_scheduler: Optional[base.ScalarOrSchedule] = None,
+  alpha_scheduler: Optional[base.ScalarOrSchedule] = None,
```

Comment:
I noticed the patter with the majority of the other classes that did not use `base.ScalarOrSchedule` as the type and tried to replicate that.

Of course, the authors in the paper specifically recommend scheduling `b3` and `alpha` because of early training instabilities, so this is a bit of a different use case.

I will make this change.